src/main/com/fulcrologic/rad/routing/html5_history.cljc

```diff
@@ -135,19 +137,18 @@
          (.back js/history))))
   (-add-route-listener! [_ listener-key f] (swap! listeners assoc listener-key f))
   (-remove-route-listener! [_ listener-key] (swap! listeners dissoc listener-key))
-  (-current-route [_] (url->route hash-based?)))
+  (-current-route [_] (url->route hash-based? prefix)))
 
-(defn html5-history
+(defn new-html5-history
   "Create a new instance of a RouteHistory object that is properly configured against the browser's HTML5 History API.
 
    `hash-based?` - Use hash-based URIs instead of paths
-   `all-events?` - Call the route listeners on all routing operations (not just pop state events)."
-  ([] (html5-history false))
-  ([hash-based?] (html5-history hash-based? false))
-  ([hash-based? all-events?]
+   `all-events?` - Call the route listeners on all routing operations (not just pop state events).
+   `prefix`      - Prepend prefix to all routes, in cases we are not running on root url (context-root)"
+  [{:keys [hash-based? all-events? prefix] :or {all-events? false, hash-based? false, prefix nil}}]
    #?(:cljs
       (try
-        (let [history            (HTML5History. hash-based? (atom {}) (atom 1) (atom 1) (atom nil) all-events?)
+        (let [history            (HTML5History. hash-based? (atom {}) (atom 1) (atom 1) (atom nil) all-events? prefix)
```

Comment:
Does prefix need a leading and/or training `/`? You should be cleaning the input so that your other expressions are making valid assumptions (e.g. making sure there is exactly one leading and (no?) trailing slash).